Support Gemini LLM and Embeddings on VertexAI https://github.com/llm-tools/embedJs/issues/46

tsibg opened the pull request:
Added support for Gemini model and embeddings on VertexAI. 
This PR also provides example and documentation on how to use, including updates to the main README.

Implemented with:
- Langchain integration for LLMS on VertexAI: https://js.langchain.com/docs/integrations/llms/google_vertex_ai
- Langchain's community integration (Experimental) for `textembedding-gecko` on VertexAI: https://js.langchain.com/docs/modules/data_connection/experimental/multimodal_embeddings/google_vertex_ai

Related Issue:
https://github.com/llm-tools/embedJs/issues/46

adhityan commented on README.md:
```diff
@@ -452,6 +454,36 @@ const ragApplication = await new RAGApplicationBuilder()
 You can read more about the various models provided by Anthropic [here](https://docs.anthropic.com/claude/docs/models-overview).
 
+
+## Vertex AI
```
I will make a few modifications to the Readme documentation post merging the PR

adhityan commented on README.md:
```diff
@@ -546,6 +578,24 @@ await new RAGApplicationBuilder()
+
+## Vertex AI Embeddings
+
+Embedding model `textembedding-gecko` with 768 dimensions on [VertexAI](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings).
```
We actually name the embedding to the actual model, not the provider (this allows us to use different classes for each model from a single provider). Could you rename the embedding model to `GeckoEmbedding` ?

tsibg replied:
Refactored with https://github.com/llm-tools/embedJs/pull/47/commits/605fbd9029c8ef05bec58edddceb956a2ba74070 
Also updated the documentation to follow the changes.

adhityan commented on package.json:
```diff
@@ -56,11 +56,13 @@
         "axios": "^1.6.8",
         "confluence.js": "^1.7.4",
         "debug": "^4.3.4",
+        "google-auth-library": "^8.9.0",
```
I would move this to optional dependency like what we have with the vector DBs. I will do this post merge, nothing for you to do here.

tsibg replied:
I removed the dependency from package.json as https://github.com/llm-tools/embedJs/pull/47/commits/e41762fbc38b15ccf4dbf623939bb2ae810a9d58 .

adhityan commented:
Thanks for sending in the PR; it's quite well written. There is one actionabale change that I think should be done pre-merge -

1. Rename the embedding model to reflect the actual model

Apart from that, there is an update I would like to do on the dependency front. I will do this and publish a new version post merge. What do you think about the change above?

tsibg commented:
> Thanks for sending in the PR; it's quite well written. There is one actionabale change that I think should be done pre-merge -
> 
> 1. Rename the embedding model to reflect the actual model
> 
> Apart from that, there is an update I would like to do on the dependency front. I will do this and publish a new version post merge. What do you think about the change above?

Thanks for the review.
1. Makes sense. Will need 1-2days to get back on this.
2. The mentioned dependency  comes from the documentation for the Embeddings' module on Langchain. Let me double check if it is really needed and how the module is using it and will update you on this matter.

tsibg commented:
@adhityan Just to summarize the changes:

1. Renamed the model to GeckoEmbedings and updated the corresponding example and READMEs
2. Bit tricky. I tried to follow the dependencies of the related LangChain modules and it differs for the Model and for the Embeddings as follows:
- The Model already adds it as peer dependency: `@langchain/google-vertexai` -> `@langchain/google-gauth` -> `google-auth-library`
- The Embedding *does not* install it - it relies on experimental module, part of `@langchain/community`. However it needs it.

tl;dr I suppose that we can safely skip it for now. We already have installed `@langchain/google-vertexai` (and its peer dependencies). 
--
Please, take a look at the changes, it should be ready for merge.

adhityan commented:
Thank you. LGTM. I will merge it now. Changes will be published to the next library version by end of week.
